2.4/root/usr/share/container-scripts/mongodb/common.sh

```diff
@@ -113,40 +113,23 @@ function mongo_initiate() {
   mongo admin --eval "${config};rs.initiate(config);${mongo_wait}"
 }
 
-# get the address of the current primary member
-function mongo_primary_member_addr() {
-  local rc=0
-
-  endpoints | grep -v "$(container_addr)" |
-  (
-    while read mongo_node; do
-      cmd_output="$(mongo admin -u admin -p "$MONGODB_ADMIN_PASSWORD" --host "$mongo_node:$CONTAINER_PORT" --eval 'print(rs.isMaster().primary)' --quiet || true)"
-
-      # Trying to find IP:PORT in output and filter out error message because mongo prints it to stdout
-      ip_and_port_regexp='[0-9]\+\.[0-9]\+\.[0-9]\+\.[0-9]\+:[0-9]\+'
-      if addr="$(echo "$cmd_output" | grep -x "$ip_and_port_regexp")"; then
-        echo -n "$addr"
-        exit 0
-      fi
-
-      echo >&2 "Cannot get address of primary from $mongo_node node: $cmd_output"
-    done
-
-    exit 1
-  ) || rc=$?
-
-  if [ $rc -ne 0 ]; then
-    echo >&2 "Cannot get address of primary node: after checking all nodes we don't have the address"
+# replset_addr return the address of the current replSet
+function replset_addr() {
+  local current_endpoints
+  current_endpoints="$(endpoints)"
+  if [ -z "${current_endpoints}" ]; then
+    echo >&2 "Cannot get address of replica set: no nodes are listed in service"
     return 1
   fi
+  echo "${MONGODB_REPLICA_NAME}/$(echo ${current_endpoints} | sed -e 's| |:27017,|g' -e 's|$|:27017|')"
```

Comment:
Looking at the docs, we see that the PORT is optional. Since the point of the PR is to simplify matters, we can simplify even further by not using `sed` to insert the default port number in hard-coded form.

Let's keep in mind that the hard-coded port number is really an arbitrary limitation, but one that makes sense in terms of being pragmatic. One could run their mongod servers in any port using OpenShift/Docker. But handling arbitrary ports would require furthers complications to the Bash script.

All that is left then is to replace all spaces with comma?

```bash
  echo "${MONGODB_REPLICA_NAME}/${current_endpoints//[[:space:]]/,}"
```

`dig` returns multiple IPs in separate lines, and the `[[:space:]]` pattern captures that (while also handling spaces, tabs, etc).